packages/adapter-surrealdb/src/index.ts

```diff
@@ -0,0 +1,336 @@
+import type Surreal from "surrealdb.js"
+import type { Result } from "surrealdb.js"
+import type {
+  Adapter,
+  AdapterUser,
+  AdapterAccount,
+  AdapterSession,
+  VerificationToken,
+} from "next-auth/adapters"
+import type { ProviderType } from "next-auth/providers"
+
+type Document = Record<string, string | null | undefined> & { id: string }
+export type UserDoc = Document & { email: string }
+export type AccountDoc<T = string> = {
+  id: string
+  userId: T
+  refresh_token?: string
+  access_token?: string
+  type: ProviderType
+  provider: string
+  providerAccountId: string
+  expires_at?: number
+}
+export type SessionDoc<T = string> = Document & { userId: T }
+
+const extractId = (surrealId: string) => surrealId.split(":")[1] ?? surrealId
+
+// Convert DB object to AdapterUser
+export const docToUser = (doc: UserDoc): AdapterUser => ({
+  ...doc,
+  id: extractId(doc.id),
+  emailVerified: doc.emailVerified ? new Date(doc.emailVerified) : null,
+})
+
+// Convert DB object to AdapterAccount
+export const docToAccount = (doc: AccountDoc): AdapterAccount => {
+  const account = {
+    ...doc,
+    id: extractId(doc.id),
+    userId: doc.userId ? extractId(doc.userId) : "",
+  }
+  return account
+}
+
+// Convert DB object to AdapterSession
+export const docToSession = (
+  doc: SessionDoc<string | UserDoc>
+): AdapterSession => ({
+  userId: extractId(
+    typeof doc.userId === "string" ? doc.userId : doc.userId.id
+  ),
+  expires: new Date(doc.expires ?? ""),
+  sessionToken: doc.sessionToken ?? "",
+})
+
+// Convert AdapterUser to DB object
+const userToDoc = (
+  user: Omit<AdapterUser, "id"> | Partial<AdapterUser>
+): Omit<UserDoc, "id"> => {
+  const doc = {
+    ...user,
+    emailVerified: user.emailVerified?.toISOString(),
+  }
+  return doc
+}
+
+// Convert AdapterAccount to DB object
+const accountToDoc = (account: AdapterAccount): Omit<AccountDoc, "id"> => {
+  const doc = {
+    ...account,
+    userId: `user:${account.userId}`,
+  }
+  return doc
+}
+
+// Convert AdapterSession to DB object
+export const sessionToDoc = (
+  session: AdapterSession
+): Omit<SessionDoc, "id"> => {
+  const doc = {
+    ...session,
+    expires: session.expires.toISOString(),
+  }
+  return doc
+}
+
+export function SurrealDBAdapter(
+  client: Promise<Surreal>
+  // options = {}
+): Adapter {
+  return {
+    async createUser(data) {
+      const surreal = await client
+      const doc = userToDoc(data)
+      const user = (await surreal.create("user", doc)) as UserDoc
+      return docToUser(user)
+    },
+    async getUser(id: string) {
+      const surreal = await client
+      try {
+        const users = (await surreal.select(`user:${id}`)) as UserDoc[]
```

Comment:
use query variables

Reply:
`surreal.select` is a syntactic sugar for `surreal.query`.

https://surrealdb.com/docs/integration/libraries/nodejs#select